src/errors/InvalidResourceError.ts

```diff
@@ -5,10 +5,19 @@
 
 // eslint-disable-next-line import/prefer-default-export
 export class InvalidResourceError extends Error {
+    readonly isInvalidResourceError: boolean;
+
     constructor(message = 'Invalid Resource') {
         // Node Error class requires passing a string message to the parent class
         super(message);
         Object.setPrototypeOf(this, InvalidResourceError.prototype);
+        this.isInvalidResourceError = true;
         this.name = this.constructor.name;
     }
 }
+export function isInvalidResourceError(err: InvalidResourceError | unknown): err is InvalidResourceError {
```

Comment:
```suggestion
export function isInvalidResourceError(err: unknown): err is InvalidResourceError {
```
If you already have a var of type InvalidResourceError then there's no point in calling this function

Reply:
The problem we are trying to solve is that we know that the err is of type InvalidResourceError but we do not know if it is of type InvalidResourceError@1.0.0 or InvalidResourceError@1.0.1. So we will need to call this function ignore the version of this error

Regardless declaring the type here as InvalidResourceError is not needed -- removing

Reply:
In general the problem to solve is vending type guards for our custom errors. The version mismatch thing was just a specific bug that was happening to us because we were relying on `instanceof` for type checking.

Actually, you may want to rename the commit to something like `feat: add custom errors type guards`